src/main/java/com/exactpro/th2/fix/client/LogImpl.java

```diff
@@ -96,7 +96,11 @@ public void onErrorEvent(String text, Throwable e) {
     private void onMessage(String message, Direction direction) throws IOException {
         Supplier<Long> sequence = direction == Direction.FIRST ? inputSeq : outputSeq;
         QueueAttribute attribute = direction == Direction.FIRST ? QueueAttribute.FIRST : QueueAttribute.SECOND;
-        messageRouter.send(MessageUtil.toBatch(message.getBytes(), connectionID, direction, sequence.get()), attribute.toString());
+        var messageGroupBatch = MessageUtil.toBatch(message.getBytes(), connectionID, direction, sequence.get());
+        messageRouter.send(messageGroupBatch, attribute.toString());
+
+        String parentEventID = FixMessage.getParentEventIDs().get(message) == null ? parentEventId : FixMessage.getParentEventIDs().get(message).getId();
```

Comment:
> ? parentEventId

I think we shouldn't produce sent-event for messages without parent-event-ids (at least we don't that in dirty-conn)